dhis-2/dhis-support/dhis-support-hibernate/src/main/java/org/hisp/dhis/config/DataSourceConfig.java

```diff
@@ -97,6 +113,34 @@ public DataSource dataSource(DhisConfigurationProvider config) {
     return createLoggingDataSource(config, actualDataSource(config));
   }
 
+  @Bean
+  public DataSource readOnlyDataSource(
+      DhisConfigurationProvider config, @Qualifier("actualDataSource") DataSource dataSource) {
+    if (StringUtils.isBlank(config.getProperty(ConfigurationKey.READ_ONLY_CONNECTION_URL))) {
+      return dataSource;
+    }
+
+    DbPoolConfig dbPoolConfig =
+        DbPoolConfig.builder()
+            .dhisConfig(config)
+            .jdbcUrl(config.getProperty(ConfigurationKey.READ_ONLY_CONNECTION_URL))
+            .username(config.getProperty(ConfigurationKey.CONNECTION_USERNAME))
```

Comment:
Is the `CONNECTION_USERNAME` and `CONNECTION_PASSWORD` used only by the read only connection or also by the default DB?

If they are the same, what if the read only DB has different username and password?
If they are not the same, would it make sense to rename them to `READ_ONLY_CONNECTION_USERNAME` and `READ_ONLY_CONNECTION_PASSWORD`?